### ISSUE TITLE
Completion based on SDK

### ISSUE THREAD

**Issue by fimkap** [maralla/completor-swift#1]
How do I override the swift completion based on the default SDK? I cannot find a way to provide it (the path for the sdk) via arguments.

**Comment by fimkap** [comment]
I have looked at the plugin's code and found that there's .sourcekitten_complete file. But when adding the arguments to this file (tried many variations) I was getting the error - using a list instead of String. A small change in completor_swift.py fixes that problem. I'm not sure that this is correct way to fix it.
```python
            file_args = self.parse_config(self.args_file)
            if file_args:
                args_str = file_args.pop()
                args.extend(['-compiler-args', args_str])
```

With this fix I can give the arguments like sdk, target and source files. The completion seems to work well and has a great potential.
The next task will be to enable completion not only after the dot but for keywords and types, module names etc. (more like in Xcode) which involve changing regular expression. But this I will ask in the main completor repository. This issue can be closed.

**Comment by maralla** [comment]
I pushed a commit to fix the `-compiler-args` bug. And it's also possible to complete keywords and types now.

**Comment by fimkap** [comment]
That's great! Thank you for the amazing plugin, the only one for swift completion that worked out of the box, fast and doesn't slow down your typing.
I performed some initial test on your latest changes - looks very good so far. A few questions though:
1. Any way to leave TAB for completion and use it for jumping to placeholders in normal mode only?
2. When I started to work on adding completion for types etc., I have just added the space to the trigger and pattern expression and this way I preserved running the usual completion options from the beginning of line so that I can use snippets. After your changes, I'm getting SourceKit completion from the beginning of line (which is perfect) but am losing the snippets from Ultisnips. Is there any way to mix SourceKit with snippets completion?
3. This one was asked in the main completion repo but wasn't answered yet. Any way to disable a popup refresh on every character? Since SourceKit uses one completion point anyway and is deterministic I see no reason to send a request on every character.
4. Any way to display the popup below the completion line for SourceKit completion as it is for other types? For some reason now it is displayed above.
